packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -248,27 +259,94 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
             visualizeElements(hilitedIds, stringToColor(row.id ?? ""));
             allIds = allIds.concat(hilitedIds);
           } else {
-            const ids: string[] = await fetchIdsFromQuery(
-              row.query ?? "",
-              iModelConnection,
-            );
-            const hiliteIds = await visualizeElementsById(
-              ids,
-              stringToColor(row.id ?? ""),
-              iModelConnection,
-            );
-            hilitedElements.current.set(row.query ?? "", hiliteIds);
+            try {
+              const ids: string[] = await fetchIdsFromQuery(
+                row.query ?? "",
+                iModelConnection,
+              );
+              const hiliteIds = await visualizeElementsById(
+                ids,
+                stringToColor(row.id ?? ""),
+                iModelConnection,
+              );
+              hilitedElements.current.set(row.query ?? "", hiliteIds);
 
-            allIds = allIds.concat(ids);
+              allIds = allIds.concat(ids);
+            } catch {
+              const index = groups.findIndex((group) => group.id === row.id);
+              setSelectedRows((rowIds) => {
+                const selectedRowIds = { ...rowIds };
+                delete selectedRowIds[index];
+                return selectedRowIds;
+              });
+              toaster.negative(`Could not load ${row.groupName}.`);
+
+            }
           }
         }
         await zoomToElements(allIds);
         setLoadingQuery(false);
       }
     },
-    [iModelConnection],
+    [iModelConnection, groups],
+  );
+
+  const controlledState = useCallback(
+    (state) => {
+      return {
+        ...state,
+        selectedRowIds: { ...selectedRows },
+      };
+    },
+    [selectedRows],
   );
 
+  const propertyMenuGoBack = useCallback(async () => {
+    clearEmphasizedElements();
+    setGroupsView(GroupsView.GROUPS);
+    await refresh();
+  }, [refresh]);
+
+  const tableStateReducer = (
+    newState: TableState,
+    action: ActionType,
+    _previousState: TableState,
+    instance?: TableInstance,
+  ): TableState => {
+    switch (action.type) {
+      case actions.toggleRowSelected: {
+        const newSelectedRows = {
+          ...selectedRows,
+        };
+        if (action.value) {
+          newSelectedRows[action.id] = true;
+        } else {
+          delete newSelectedRows[action.id];
```

Comment:
another one

Reply:
Sure, I was just reimplementing the select logic from react-table itself. 
https://github.com/TanStack/react-table/blob/76a4a861ee56b782404ef91987c3b5691ecf2ebc/src/plugin-hooks/useRowSelect.js#L161

Reply:
I might have to fight Tanner :P